controller/Delivery.php

```diff
@@ -0,0 +1,432 @@
+                    Breadcrumbs::deliveryTestTaker($testCenter, $delivery)
+                )
+            );
+
+        } catch (ServiceNotFoundException $e) {
+            \common_Logger::w('No delivery service defined for proctoring');
+            $this->returnError('Proctoring interface not available');
+        }
+
+    }
+
+    /**
+     * Gets the value of a string property from a user
+     * @param User $user
+     * @param string $property
+     * @return mixed|string
+     */
+    private function getUserStringProp($user, $property) {
+        $value = $user->getPropertyValues($property);
+        return empty($value) ? '' : current($value);
+    }
+
+    /**
+     * Get a list of test takers usable in a table page
+     * @param User[] $users
+     * @param array $options
+     * @return array
+     */
+    private function getTestTakersPage($users, $options) {
+        $amount = count($users);
+        $rows = max(1, abs(ceil(isset($options['rows']) ? $options['rows'] : 25)));
+        $total = ceil($amount / $rows);
+        $page = max(1, floor(min(isset($options['page']) ? $options['page'] : 1, $total)));
+        $start = ($page - 1) * $rows;
+        $list = array();
+
+        $users = array_slice($users, ($page - 1) * $rows, $rows);
+
+        foreach($users as $user) {
+            /* @var $user User */
+            $firstName = $this->getUserStringProp($user, PROPERTY_USER_FIRSTNAME);
+            $lastName = $this->getUserStringProp($user, PROPERTY_USER_LASTNAME);
+            if (empty($firstName) && empty($lastName)) {
+                $firstName = $this->getUserStringProp($user, RDFS_LABEL);
+            }
+
+            $list[] = array(
+                'id' => $user->getIdentifier(),
+                'firstname' => $firstName,
+                'lastname' => $lastName,
+                'company' => '',
+                'status' => ''
+            );
+        }
+
+        return array(
+            'offset' => $start,
+            'length' => count($list),
+            'amount' => $amount,
+            'total'  => $total,
+            'page'   => $page,
+            'rows'   => $rows,
+            'data'   => $list
+        );
+    }
+
+    /**
+     * Gets the list of test takers assigned to a delivery
+     *
+     * @throws \common_Exception
+     * @throws \oat\oatbox\service\ServiceNotFoundException
+     */
+    public function deliveryTestTakers() {
+
+        $deliveryId = $this->getRequestParameter('id');
+
+        try {
+
+//            $deliveryService = $this->getServiceManager()->get('taoProctoring/delivery');
+//
+//            $requestOptions = $this->getRequestOptions();
+//            $users = $deliveryService->getDeliveryTestTakers($deliveryId, $requestOptions);
+//            $testTakers = $this->getTestTakersPage($users, $requestOptions);
+   
+            //@TODO mock list of test taker
+            $this->returnJson(array());
```

Comment:
I have to temporarily disable the delivery test taker listing here because of the incompatibility with the dummy mock data